ext/mysqlnd/mysqlnd_wireprotocol.c

```diff
@@ -540,15 +539,14 @@ size_t php_mysqlnd_auth_write(MYSQLND_CONN_DATA * conn, void * _packet)
 			DBG_RETURN(0);
 		}
 		if (packet->auth_data_len) {
-			memcpy(p, packet->auth_data, packet->auth_data_len);
-			p+= packet->auth_data_len;
+			p = zend_mempcpy(p, packet->auth_data,
+					 packet->auth_data_len);
```

Comment:
```suggestion
			p = zend_mempcpy(p, packet->auth_data, packet->auth_data_len);
```
Seems more logical, or is there a reason for the line break?